### PR TITLE
Session file security and logging

### PR DESCRIPTION
**Pull request by GawainLynch**
* Make sure we respect `session.save-path` settings for file permissions
* Log errors handling session files

### CODE REVIEW

File: src/Session/Handler/FileHandler.php [hunk 1]
```diff
@@ -122,6 +152,7 @@ public function gc($maxlifetime)
             try {
                 $this->fs->remove($file);
             } catch (IOException $e) {
+                $this->logger->error('Unable to remove session file ' . $file);
                 return false;
```
Comment (CarsonF) [inline src/Session/Handler/FileHandler.php]: @GawainLynch Can you delete this return line? we shouldn't be stopping for one errored file.

File: src/Session/Handler/FileHandler.php [hunk 2]
```diff
@@ -13,22 +15,33 @@
...
     {
         $this->fs = $filesystem ?: new Filesystem();
+        $this->logger = $logger ?: new NullLogger();
 
+        // @see http://php.net/manual/en/session.configuration.php#ini.session.save-path
+        $depth = 1;
+        $mode = 0600;
         $savePath = $savePath ?: sys_get_temp_dir();
 
         // Handle BC 'N;/path' and 'N;octal-mode;/path` which are not supported here
```
Comment (CarsonF) [inline src/Session/Handler/FileHandler.php]: They aren't supported here? ;)

File: src/Session/Handler/FileHandler.php (hunk 1 above)
Comment (GawainLynch) [inline src/Session/Handler/FileHandler.php]: So we should always return `true`?

File: src/Session/Handler/FileHandler.php [hunk 3]
```diff
@@ -13,22 +15,33 @@
...
-     * @param string     $savePath Path of directory to save session files.
-     *                             Default null will leave setting as defined by system temp directory.
-     * @param Filesystem $filesystem
+     * @param string          $savePath   Path of directory to save session files.
+     *                                    Default null will leave setting as defined by system temp directory.
+     * @param Filesystem      $filesystem
+     * @param LoggerInterface $logger
      */
-    public function __construct($savePath = null, Filesystem $filesystem = null)
+    public function __construct($savePath = null, Filesystem $filesystem = null, LoggerInterface $logger = null)
```
Comment (CarsonF) [inline src/Session/Handler/FileHandler.php]: Can we flip logger and filesystem? We are way more likely to want to pass in a logger vs a filesystem

File: src/Session/Handler/FileHandler.php (hunk 2 above)
Comment (GawainLynch) [inline src/Session/Handler/FileHandler.php]: Hang on, let me `git blame` that for you ;-)

**Comment by CarsonF** [comment]
Nice work mate :+1:

File: src/Session/Handler/FileHandler.php (hunk 1 above)
Comment (CarsonF) [inline src/Session/Handler/FileHandler.php]: Doesn't matter. SessionStorage doesn't check the returns. Idk what I would do with a false return.

File: src/Session/Handler/FileHandler.php (hunk 2 above)
Comment (CarsonF) [inline src/Session/Handler/FileHandler.php]: Well they weren't until now when you added support for them ha

**Pull request merged**